File: codeforces/problemset/A. Pashmak and Garden.cpp
#include<bits/stdc++.h>
using namespace std;
struct Var{
    int x;
    int y;
};
int main(){
    Var a, b;
    scanf("%d %d %d %d",&a.x,&a.y,&b.x,&b.y);
    int diffY = a.y - b.y, diffX = a.x - b.x;
    if(diffY * -1 == 0){
        printf("%d %d ",a.x,a.y+abs(diffX));
        printf("%d %d ",b.x,b.y+abs(diffX));
    }else if(diffX * -1 == 0){
        printf("%d %d ",a.x + diffY,a.y);
        printf("%d %d ",b.x + diffY,b.y);
    }else if(a.y - b.y == a.x - b.x or a.y - b.y == b.x - a.x){
        printf("%d %d ",a.x ,b.y);
        printf("%d %d ",b.x ,a.y);
    }else{
        puts("-1");
    }
    return 0;
}

File: codeforces/problemset/A. Pashmak and Garden.java
import java.util.Scanner;

class Var {
    int x;
    int y;
}

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Initialize the two Var objects
        Var a = new Var();
        Var b = new Var();
        
        // Read input values
        a.x = scanner.nextInt();
        a.y = scanner.nextInt();
        b.x = scanner.nextInt();
        b.y = scanner.nextInt();
        
        int diffY = a.y - b.y;
        int diffX = a.x - b.x;

        // Conditions based on the differences between points
        if (diffY * -1 == 0) {
            System.out.printf("%d %d ", a.x, a.y + Math.abs(diffX));
            System.out.printf("%d %d ", b.x, b.y + Math.abs(diffX));
        } else if (diffX * -1 == 0) {
            System.out.printf("%d %d ", a.x + diffY, a.y);
            System.out.printf("%d %d ", b.x + diffY, b.y);
        } else if (a.y - b.y == a.x - b.x || a.y - b.y == b.x - a.x) {
            System.out.printf("%d %d ", a.x, b.y);
            System.out.printf("%d %d ", b.x, a.y);
        } else {
            System.out.println("-1");
        }

        scanner.close();
    }
}
